gi_wrapper_cpp/wrapper.cpp

```diff
@@ -20,70 +20,80 @@
 
 #include <iostream>
 #include <cstdlib>
-#include <imebra/imebra.h>
+#include "gdcmReader.h"
+#include "gdcmGlobal.h"
+#include "gdcmDicts.h"
+#include "gdcmDict.h"
+#include "gdcmAttribute.h"
+#include "gdcmStringFilter.h"
+#include "gdcmImageReader.h"
+#include "gdcmImageWriter.h"
+
 #include "dicom_reader.h"
 #include "dicom_writer.h"
 #include "wrapper_logger.h"
 
-using namespace imebra;
+using namespace gdcm;
 using namespace std;
 
 
-DataSet* loadDicom(string filepath)
+File& loadDicom(string filepath)
 {
     logWrapperMessage("Loading DICOM file: " + filepath);
-    return CodecFactory::load(filepath);
+    Reader reader;
+    reader.SetFileName(filepath.c_str());
+    return reader.GetFile();
```

Comment:
@kdbogusz that is a potential access to uninitialized memory - gdcm documentation doesn't specify ownership rules to the returned file thus I would assume that it is owned by the reader object thus after exiting loadDicom scope the referenced file object returned by this function is destroyed (as the reader is destroyed) - because of that I would suggest returning the file object by value (File loadDicom(...)) or copy that to heap and return as unique_ptr (less preferable)